apps/studio/components/interfaces/APIKeys/SecretAPIKeys.tsx

```diff
@@ -0,0 +1,146 @@
+      </TableCell>
+    </TableRow>
+  )
+
+  const TableContainer = ({ children }: { children: React.ReactNode }) => (
+    <div>
+      <FormHeader
+        title="Secret keys"
+        description="These API keys allow privileged access to your project's APIs. Use in servers, functions, workers or other backend components of your application. Keep secret and never publish."
+        actions={<CreateSecretAPIKeyModal />}
+      />
+      <Card className={cn('w-full overflow-hidden', !empty && 'bg-surface-100')}>
+        <CardContent className="p-0">
+          <Table className="p-5">
+            <TableHeader>
+              <TableRow className={cn('bg-200', empty && 'hidden')}>
+                <TableHead
+                  key=""
+                  className="text-left font-mono uppercase text-xs text-foreground-lighter h-auto py-2 overflow-hidden"
+                >
+                  Name
+                </TableHead>
+                <TableHead className="text-left font-mono uppercase text-xs text-foreground-lighter h-auto py-2 pr-0">
+                  API Key
+                </TableHead>
+                <TableHead
+                  className="text-right font-mono uppercase text-xs text-foreground-lighter h-auto py-2"
+                  key="actions"
+                />
+              </TableRow>
+            </TableHeader>
+            <TableBody className="">{children}</TableBody>
+          </Table>
+        </CardContent>
+      </Card>
+    </div>
+  )
+
+  if (isLoadingApiKeys || isLoadingPermissions) {
+    return (
+      <TableContainer>
+        <RowLoading />
+        <RowLoading />
+      </TableContainer>
+    )
+  }
+
+  if (!canReadAPIKeys) {
+    return (
+      <TableContainer>
+        <div className="!rounded-b-md overflow-hidden py-12 flex flex-col gap-1 items-center justify-center">
+          <EyeOffIcon />
+          <p className="text-sm text-foreground">
+            You do not have permission to read API Secret Keys
+          </p>
+          <p className="text-foreground-light">
+            Contact your organization owner/admin to request access.
+          </p>
+        </div>
+      </TableContainer>
+    )
+  }
+
+  if (error) {
+    return (
+      <TableContainer>
+        <div className="!rounded-b-md overflow-hidden py-12 flex flex-col gap-1 items-center justify-center">
+          <WarningIcon />
+          <p className="text-sm text-warning-600">Error loading Secret API Keys</p>
+          <p className="text-warning/75">{error.message}</p>
+        </div>
+      </TableContainer>
+    )
+  }
+
+  if (empty) {
+    return (
+      <TableContainer>
+        <div className="!rounded-b-md overflow-hidden py-12 flex flex-col gap-1 items-center justify-center">
+          <p className="text-sm text-foreground">No secret API keys exist</p>
+          <p className="text-sm text-foreground-light">
+            Your project can't be accessed using secret API keys.
```

Comment:
nit: this description here is quite vague RE what access means? does it mean the database can't be access with the secret keys? depending on what it means, i think it'll be important to highlight as users seeing this new concept may have some expectations of what this new secret API key does due to the previous implementation

Reply:
follow up